### PR TITLE
support peer dependencies

### PR DESCRIPTION
**Pull request by davidfirst**

### CODE REVIEW

File: e2e/flows/peer-dependencies.e2e.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    helper.addComponentBarFoo();
+    helper.commitComponentBarFoo();
+  });
+  it('should save the peer dependencies in the model', () => {
+    const output = helper.catComponent('bar/foo@latest');
+    expect(output).to.have.property('peerPackageDependencies');
+    expect(output.peerPackageDependencies).to.have.property('chai');
+    expect(output.peerPackageDependencies.chai).to.equal('>= 2.1.2 < 5');
+  });
+  it('bit show should display the peer dependencies', () => {
```
Comment (GiladShoham) [inline e2e/flows/peer-dependencies.e2e.js]: I'm thinking maybe it's better to put this case in show.e2e?

File: e2e/flows/peer-dependencies.e2e.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    helper.createComponentBarFoo();
+    helper.addComponentBarFoo();
+    helper.commitComponentBarFoo();
+  });
+  it('should save the peer dependencies in the model', () => {
+    const output = helper.catComponent('bar/foo@latest');
+    expect(output).to.have.property('peerPackageDependencies');
+    expect(output.peerPackageDependencies).to.have.property('chai');
+    expect(output.peerPackageDependencies.chai).to.equal('>= 2.1.2 < 5');
```
Comment (GiladShoham) [inline e2e/flows/peer-dependencies.e2e.js]: Check also that isn't in the regular / dev deps

File: src/consumer/component/dependencies-resolver.js [hunk 3]
```diff
@@ -325,6 +326,19 @@ Try to run "bit import ${componentId} --objects" to get the component saved in t
   return { componentsDeps, devComponentsDeps, packagesDeps, devPackagesDeps, untrackedDeps, relativeDeps, missingDeps };
 }
 
+function findPeerDependencies(consumerPath: string): Object {
```
Comment (GiladShoham) [inline src/consumer/component/dependencies-resolver.js]: this logic should be in bit-js in the build-tree file.
There is also some logic related to peerDeps (i'm not sure that alone will be enough for you), but anyway, the correct place is there for sure.

File: e2e/flows/peer-dependencies.e2e.js (hunk 1 above)
Comment (davidfirst) [reply e2e/flows/peer-dependencies.e2e.js]: I use the e2e/flow directory for testing an entire flow. In this case, once the feature is done, I'll be testing more scenario. (you can see other e2e tests in this directory for examples).

File: src/consumer/component/dependencies-resolver.js (hunk 3 above)
Comment (davidfirst) [reply src/consumer/component/dependencies-resolver.js]: I agree. It just takes too much time to work with bit-javascript. I'll add a todo comment to move it there once we have the mono repo.

**Pull request merged**